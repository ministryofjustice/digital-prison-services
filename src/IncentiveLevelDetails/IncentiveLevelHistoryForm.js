import React from 'react'
import PropTypes from 'prop-types'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import Button from '@govuk-react/button'
import moment from 'moment'
import { connect } from 'react-redux'
import { Form } from 'react-final-form'
import ErrorText from '@govuk-react/error-text'
import ErrorSummary from '@govuk-react/error-summary'
import FormDatePicker from '../DatePickers/FormDatePicker'
import { linkOnClick } from '../utils'

import { FullWidthSelect, SearchArea, FiltersLabel, DateRangeLabel } from '../Components/FilterForm/FilterForm.styles'
import { FieldWithError, onHandleErrorClick } from '../final-form-govuk-helpers'
import validateThenSubmit from '../Components/FilterForm/FilterFormValidation'

export const IepHistoryForm = ({ establishments, levels, search, fieldValues, reset, now }) => (
  <Form
    initialValues={fieldValues}
    onSubmit={values => {
      const errors = validateThenSubmit({
        fromDate: values.fromDate,
        toDate: values.toDate,
        submit: search,
      })(values)
      if (errors) window.scrollTo(0, 0)
      return errors
    }}
    render={({ error, handleSubmit, submitError, submitting, form, values }) => (
      <>
        {error && <ErrorText> {error} </ErrorText>}
        {submitError && (
          <ErrorSummary onHandleErrorClick={onHandleErrorClick} heading="There is a problem" errors={submitError} />
        )}
        <form onSubmit={handleSubmit}>
          <SearchArea>
            <FormFields
              now={now}
              errors={submitError}
              submitting={submitting}
              establishments={establishments}
              levels={levels}
              values={values}
              reset={() => {
                reset().then(() => form.reset({}))
              }}
            />
          </SearchArea>
        </form>
      </>
    )}
  />
)

const shouldShowDay = now => date => date.isSameOrBefore(now, 'day')

export const FormFields = ({ now, errors, establishments, levels, submitting, values, reset }) => (
  <>
    <GridRow>
      <GridCol setDesktopWidth="one-half">
        <GridRow>
          <FiltersLabel>Filters</FiltersLabel>
        </GridRow>
        <GridRow>
          <GridCol>
            <FieldWithError
              errors={errors}
              name="establishment"
              component={FullWidthSelect}
              label="Establishment"
              id="establishment-select"
              value={values.iepEstablishment}
            >
              <option value="">All</option>
              {establishments.map(estb => (
                <option key={estb.agencyId} value={estb.agencyId}>
                  {estb.description}
                </option>
              ))}
            </FieldWithError>
          </GridCol>
          <GridCol>
            <FieldWithError
              errors={errors}
              name="level"
              component={FullWidthSelect}
              label="Incentive level"
              id="iep-level-select"
              value={values.iepLevel}
            >
              <option value="">All</option>
              {levels.map(level => (
                <option key={level} value={level}>
                  {level}
                </option>
              ))}
            </FieldWithError>
          </GridCol>
        </GridRow>
      </GridCol>
      <GridCol setDesktopWidth="one-half">
        <GridRow>
          <DateRangeLabel>Date range</DateRangeLabel>
        </GridRow>
        <GridRow>
          <GridCol id="from-date" setDesktopWidth="one-third">
            <FieldWithError
              errors={errors}
              name="fromDate"
              component={FormDatePicker}
              label="From"
              value={values.fromDate}
              shouldShowDay={shouldShowDay(now)}
            />
          </GridCol>
          <GridCol id="to-date" setDesktopWidth="one-third">
            <FieldWithError
              errors={errors}
              name="toDate"
              component={FormDatePicker}
              label="To"
              value={values.toDate}
              shouldShowDay={shouldShowDay(now)}
            />
          </GridCol>
          <GridCol setDesktopWidth="one-third">
            <p className="form-label visible-md visible-lg" />
            <Button id="apply-filter" disabled={submitting} type="submit">
              Apply filters
            </Button>
          </GridCol>
        </GridRow>
      </GridCol>
    </GridRow>
    <a id="clear-filters" className="clear-filters link clickable" {...linkOnClick(reset)}>
      Clear filters
    </a>
  </>
)

IepHistoryForm.propTypes = {
  now: PropTypes.instanceOf(moment).isRequired,
  search: PropTypes.func.isRequired,
  reset: PropTypes.func.isRequired,
  // mapStateToProps
  establishments: PropTypes.arrayOf(
    PropTypes.shape({
      agencyId: PropTypes.string.isRequired,
      agencyType: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,
  levels: PropTypes.arrayOf(PropTypes.string).isRequired,
  fieldValues: PropTypes.shape({
    establishment: PropTypes.string,
    level: PropTypes.string,
    fromDate: PropTypes.instanceOf(moment),
    toDate: PropTypes.instanceOf(moment),
  }).isRequired,
}

FormFields.propTypes = {
  now: PropTypes.instanceOf(moment).isRequired,
  reset: PropTypes.func.isRequired,
  establishments: PropTypes.arrayOf(
    PropTypes.shape({
      agencyId: PropTypes.string.isRequired,
      agencyType: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,
  values: PropTypes.shape({
    iepEstablishment: PropTypes.string,
    iepLevel: PropTypes.string,
    fromDate: PropTypes.instanceOf(moment),
    toDate: PropTypes.instanceOf(moment),
  }).isRequired,
  levels: PropTypes.arrayOf(PropTypes.string).isRequired,
  submitting: PropTypes.bool.isRequired,
  errors: PropTypes.arrayOf(PropTypes.shape({})),
}

FormFields.defaultProps = {
  errors: [],
}

const mapStateToProps = state => ({
  now: state.iepHistory.now,
  establishments: state.iepHistory.establishments,
  levels: state.iepHistory.levels,
  fieldValues: {
    establishment: state.iepHistory.establishment,
    level: state.iepHistory.level,
    fromDate: state.iepHistory.fromDate,
    toDate: state.iepHistory.toDate,
  },
})

export default connect(mapStateToProps)(IepHistoryForm)
