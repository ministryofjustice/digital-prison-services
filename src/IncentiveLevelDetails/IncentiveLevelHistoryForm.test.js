import React from 'react'
import moment from 'moment'
import testRenderer from 'react-test-renderer'

import IncentiveLevelHistoryForm from './IncentiveLevelHistoryForm'

const initialState = {
  iepHistory: {
    now: moment('2017-03-21'),
    currentIepLevel: 'Standard',
    daysOnIepLevel: 625,
    currentIepDateTime: '2017-08-15T16:04:35',
    nextReviewDate: '15/08/2018',
    establishments: [{ agencyId: 'LEI', description: 'Leeds', agencyType: 'INST' }],
    levels: ['Standard'],
    results: [
      {
        bookingId: -1,
        iepDate: '2017-08-13',
        iepTime: '2017-08-13T16:04:35',
        formattedTime: '13/08/2017 - 16:04',
        iepEstablishment: 'Leeds',
        iepStaffMember: 'Staff Member',
        agencyId: 'LEI',
        iepLevel: 'Standard',
        userId: 'ITAG_USER',
      },
    ],
  },
}

describe('Incentive level history Form', () => {
  Date.now = jest.fn(() => new Date(Date.UTC(2017, 0, 1)).valueOf())
  const store = {}
  const history = {}
  beforeEach(() => {
    store.getState = jest.fn()
    store.subscribe = jest.fn()
    store.dispatch = jest.fn()
    history.push = jest.fn()
    history.replace = jest.fn()
    store.getState.mockReturnValue(initialState)
  })

  it('should render the Incentive level history table correctly', () => {
    const wrapper = testRenderer
      .create(<IncentiveLevelHistoryForm store={store} search={jest.fn()} reset={jest.fn()} />)
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
