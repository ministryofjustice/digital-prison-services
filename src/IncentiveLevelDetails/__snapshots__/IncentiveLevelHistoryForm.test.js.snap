// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incentive level history Form should render the Incentive level history table correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-chPdSV ddsHXB"
  >
    <div
      className="sc-bdVaJa iAgRsv"
    >
      <div
        className="sc-bwzfXH kUWxme"
      >
        <div
          className="sc-bdVaJa iAgRsv"
        >
          <h3
            className="sc-fjdhpX dWzUoi sc-kGXeez iHgsBB"
          >
            Filters
          </h3>
        </div>
        <div
          className="sc-bdVaJa iAgRsv"
        >
          <div
            className="sc-bwzfXH gWDLqX"
          >
            <label
              className="sc-VigVT iBWXYf sc-kAzzGY drBIpO"
              id="establishment-select"
            >
              <span
                className="sc-gqjmRU dSGTxK"
              >
                Establishment
              </span>
              <select
                className="sc-cSHVUG iCHBuG"
                name="establishment"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  All
                </option>
                <option
                  value="LEI"
                >
                  Leeds
                </option>
              </select>
            </label>
          </div>
          <div
            className="sc-bwzfXH gWDLqX"
          >
            <label
              className="sc-VigVT iBWXYf sc-kAzzGY drBIpO"
              id="iep-level-select"
            >
              <span
                className="sc-gqjmRU dSGTxK"
              >
                Incentive level
              </span>
              <select
                className="sc-cSHVUG iCHBuG"
                name="level"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  All
                </option>
                <option
                  value="Standard"
                >
                  Standard
                </option>
              </select>
            </label>
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH kUWxme"
      >
        <div
          className="sc-bdVaJa iAgRsv"
        >
          <span
            className="sc-kpOJdX bxPfDl"
          >
            Date range
          </span>
        </div>
        <div
          className="sc-bdVaJa iAgRsv"
        >
          <div
            className="sc-bwzfXH itECxU"
            id="from-date"
          >
            <div
              className="rdt"
            >
              <div>
                <label
                  className="sc-VigVT dRnUNG"
                >
                  <span
                    className="sc-gqjmRU dSGTxK"
                  >
                     
                    From
                     
                  </span>
                  <input
                    className="sc-jTzLTM iTXUSb form-control"
                    name="fromDate"
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={true}
                    type="text"
                    value=""
                  />
                </label>
              </div>
              <div
                className="rdtPicker"
              >
                <div
                  className="rdtDays"
                >
                  <table>
                    <thead>
                      <tr>
                        <th
                          className="rdtPrev"
                          onClick={[Function]}
                        >
                          <span>
                            ‹
                          </span>
                        </th>
                        <th
                          className="rdtSwitch"
                          colSpan={5}
                          data-value={0}
                          onClick={[Function]}
                        >
                          January 2017
                        </th>
                        <th
                          className="rdtNext"
                          onClick={[Function]}
                        >
                          <span>
                            ›
                          </span>
                        </th>
                      </tr>
                      <tr>
                        <th
                          className="dow"
                        >
                          Mo
                        </th>
                        <th
                          className="dow"
                        >
                          Tu
                        </th>
                        <th
                          className="dow"
                        >
                          We
                        </th>
                        <th
                          className="dow"
                        >
                          Th
                        </th>
                        <th
                          className="dow"
                        >
                          Fr
                        </th>
                        <th
                          className="dow"
                        >
                          Sa
                        </th>
                        <th
                          className="dow"
                        >
                          Su
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={26}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          26
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={27}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          27
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={28}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          28
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={29}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          29
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={30}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          30
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={31}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          31
                        </td>
                        <td
                          className="rdtDay rdtToday"
                          data-month={0}
                          data-value={1}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          1
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={2}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          2
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={3}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          3
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={4}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          4
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={5}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          5
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={6}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          6
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={7}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          7
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={8}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          8
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={9}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          9
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={10}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          10
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={11}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          11
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={12}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          12
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={13}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          13
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={14}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          14
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={15}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          15
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={16}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          16
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={17}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          17
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={18}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          18
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={19}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          19
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={20}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          20
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={21}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          21
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={22}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          22
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={23}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          23
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={24}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          24
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={25}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          25
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={26}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          26
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={27}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          27
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={28}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          28
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={29}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          29
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={30}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          30
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={31}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          31
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={1}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          1
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={2}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          2
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={3}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          3
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={4}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          4
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={5}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          5
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
          <div
            className="sc-bwzfXH itECxU"
            id="to-date"
          >
            <div
              className="rdt"
            >
              <div>
                <label
                  className="sc-VigVT dRnUNG"
                >
                  <span
                    className="sc-gqjmRU dSGTxK"
                  >
                     
                    To
                     
                  </span>
                  <input
                    className="sc-jTzLTM iTXUSb form-control"
                    name="toDate"
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={true}
                    type="text"
                    value=""
                  />
                </label>
              </div>
              <div
                className="rdtPicker"
              >
                <div
                  className="rdtDays"
                >
                  <table>
                    <thead>
                      <tr>
                        <th
                          className="rdtPrev"
                          onClick={[Function]}
                        >
                          <span>
                            ‹
                          </span>
                        </th>
                        <th
                          className="rdtSwitch"
                          colSpan={5}
                          data-value={0}
                          onClick={[Function]}
                        >
                          January 2017
                        </th>
                        <th
                          className="rdtNext"
                          onClick={[Function]}
                        >
                          <span>
                            ›
                          </span>
                        </th>
                      </tr>
                      <tr>
                        <th
                          className="dow"
                        >
                          Mo
                        </th>
                        <th
                          className="dow"
                        >
                          Tu
                        </th>
                        <th
                          className="dow"
                        >
                          We
                        </th>
                        <th
                          className="dow"
                        >
                          Th
                        </th>
                        <th
                          className="dow"
                        >
                          Fr
                        </th>
                        <th
                          className="dow"
                        >
                          Sa
                        </th>
                        <th
                          className="dow"
                        >
                          Su
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={26}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          26
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={27}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          27
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={28}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          28
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={29}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          29
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={30}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          30
                        </td>
                        <td
                          className="rdtDay rdtOld"
                          data-month={11}
                          data-value={31}
                          data-year={2016}
                          onClick={[Function]}
                        >
                          31
                        </td>
                        <td
                          className="rdtDay rdtToday"
                          data-month={0}
                          data-value={1}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          1
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={2}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          2
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={3}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          3
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={4}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          4
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={5}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          5
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={6}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          6
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={7}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          7
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={8}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          8
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={9}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          9
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={10}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          10
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={11}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          11
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={12}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          12
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={13}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          13
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={14}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          14
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={15}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          15
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={16}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          16
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={17}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          17
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={18}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          18
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={19}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          19
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={20}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          20
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={21}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          21
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={22}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          22
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={23}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          23
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={24}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          24
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={25}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          25
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={26}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          26
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={27}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          27
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={28}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          28
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={29}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          29
                        </td>
                      </tr>
                      <tr>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={30}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          30
                        </td>
                        <td
                          className="rdtDay"
                          data-month={0}
                          data-value={31}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          31
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={1}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          1
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={2}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          2
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={3}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          3
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={4}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          4
                        </td>
                        <td
                          className="rdtDay rdtNew"
                          data-month={1}
                          data-value={5}
                          data-year={2017}
                          onClick={[Function]}
                        >
                          5
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
          <div
            className="sc-bwzfXH itECxU"
          >
            <p
              className="form-label visible-md visible-lg"
            />
            <button
              className="sc-htpNat ljwtnZ"
              disabled={false}
              id="apply-filter"
              type="submit"
            >
              Apply filters
            </button>
          </div>
        </div>
      </div>
    </div>
    <a
      className="clear-filters link clickable"
      id="clear-filters"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Clear filters
    </a>
  </div>
</form>
`;
